apprating/src/main/java/com/mindera/skeletoid/apprating/controller/AppRatingController.kt

```diff
@@ -0,0 +1,124 @@
+    fun setupConditions(countsPerTimeInterval: Pair<Int, Long>, promptTimeInterval: Long) {
+        this.countsPerTimeInterval = countsPerTimeInterval
+        this.promptTimeInterval = promptTimeInterval
+    }
+
+    /**
+     * Checks if the data in the store matches the conditions and updates the values in the store.
+     *
+     * @param context Context
+     * @param dialogResultCallback Callback to handle default dialog response
+     * @return true if the conditions were met, and the dialog can be prompted.
+     */
+    fun promptDialog(context: Context, dialogResultCallback: AppRatingDialogResponseCallback? = null): Boolean {
+        val store = AppRatingStore(context)
+        return shouldPromptDialog(store).also {
+            if (it) {
+                dialogResultCallback?.let {
+                    TODO("IMPLEMENT DEFAULT DIALOG")
+                }
+                store.promptedCount++
+                store.lastTimePrompted = DateUtils.formatDate(Date())
+            }
+        }
+    }
+
+    /**
+     * Handles the response of the dialog by updating the values in the store.
+     * If response is RATE_LATER it should schedule a one time work that starts after a delay equal to the promptTimeInterval (in days).
+     * If response is RATE_NOW or NEVER_RATE it simply updates the store.
+     *
+     * @param context Context
+     * @param response Selected response to the dialog
+     */
+    fun handleDialogResponse(context: Context, response: AppRatingDialogResponse) {
+        val store = AppRatingStore(context)
+        when (response) {
+            AppRatingDialogResponse.RATE_NOW, AppRatingDialogResponse.NEVER_RATE -> {
+                store.alreadyRated = true
+            }
+            AppRatingDialogResponse.RATE_LATER -> {
+                promptTimeInterval?.let { AppRatingJobInitializer.schedule(it) }
+            }
+        }
+    }
+
+    /**
+     * Checks if the values in the store matches the conditions to prompt the dialog.
+     *
+     * @param store Store that has the values related with the app rating conditions
+     * @return true if the conditions were met
+     */
+    private fun shouldPromptDialog(store: AppRatingStore): Boolean {
+        return if (store.initialPromptDate.isEmpty()) {
+            store.initialPromptDate = DateUtils.formatDate(Date())
+            store.promptedCount = 0
+            store.alreadyRated = false
+            true
+        } else {
+            !store.alreadyRated &&
+                    checkCountPerTimeCondition(store) &&
+                    checkPromptTimeIntervalCondition(store)
+        }
+    }
+
+    /**
+     * Checks if already reached the maximum amount of time that the rating dialog can be shown in a certain period of time.
+     *
+     * @param store Store that has the values related with the app rating conditions
+     * @return true if the conditions were met
+     */
+    private fun checkCountPerTimeCondition(store: AppRatingStore): Boolean {
```

Comment:
feels strange to have a function called `check` that returns a boolean. Can we have a different name?